tests/suites/test_suite_pk.function

```diff
@@ -90,13 +90,128 @@ size_t mbedtls_rsa_key_len_func( void *ctx )
     return( ((const mbedtls_rsa_context *) ctx)->len );
 }
 #endif /* MBEDTLS_RSA_C */
+
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#define PK_PSA_INVALID_SLOT 0 /* guaranteed invalid */
+
+/*
+ * Generate a key in a free key slot and return this key slot,
+ * or PK_PSA_INVALID_SLOT if no slot was available.
+ * The key uses NIST P-256 and is usable for signing with SHA-256.
+ */
+psa_key_slot_t pk_psa_genkey( void )
+{
+    psa_key_slot_t key;
+
+    const int curve = PSA_ECC_CURVE_SECP256R1;
+    const psa_key_type_t type = PSA_KEY_TYPE_ECC_KEYPAIR(curve);
+    const size_t bits = 256;
+
+    psa_key_policy_t policy;
+
+    /* find a free key slot
+     * this is temporary, a new PSA API for this is coming */
+    for( key = 1; key <= 32; key++ )
```

Comment:
Just a reminder that this too should be shared with @AndrzejKurek's code from #2159.

Reply:
Will be addressed when rebasing on #2184.